### PR TITLE
[cmake] add imx support

### PR DESCRIPTION
**Pull request by lrusak**
## Description

Add imx support to cmake

This supercedes #10563 

I'm not sure if we want to use IMX or IMXVPU

Backport is available here, https://github.com/lrusak/xbmc/commit/6d60b5046861ab844c055b25b2ba2da2b8b323e9

## Motivation and Context

So kodi can be built for imx SOC's

## How Has This Been Tested?

In LibreELEC

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by lrusak** [comment]
ping @mk01 

ping cmake people @fetzerch @hudokkow

**Comment by mk01** [comment]
@lrusak 
looks to be working

File: cmake/modules/FindIMX.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+find_package_handle_standard_args(IMX
+                                  REQUIRED_VARS IMX_INCLUDE_DIR FSLVPUWRAP_LIBRARY VPU_LIBRARY G2D_LIBRARY)
+
+if(IMX_FOUND)
+  set(IMX_INCLUDE_DIRS ${IMX_INCLUDE_DIR})
+  set(IMX_LIBRARIES ${FSLVPUWRAP_LIBRARY} ${VPU_LIBRARY} ${G2D_LIBRARY})
+  set(IMX_DEFINITIONS -DHAS_IMXVPU=1 -DLINUX -DEGL_API_FB)
+endif()
+
+mark_as_advanced(IMX_INCLUDE_DIR IMX_LIBRARIES IMX_DEFINITIONS)
```
Comment (fetzerch) [inline cmake/modules/FindIMX.cmake]: mark_as_advanced(IMX_INCLUDE_DIR FSLVPUWRAP_LIBRARY VPU_LIBRARY G2D_LIBRARY)

**Review by fetzerch — COMMENTED** [review]
One small issue, otherwise looks fine to me. thanks

**Comment by lrusak** [comment]
thanks @fetzerch. updated.

**Review by fetzerch — APPROVED** [review]
looks good, thanks

jenkins build this please

**Comment by fetzerch** [comment]
jenkins build this please

**Pull request merged**